src/session.rs

```diff
@@ -53,13 +53,107 @@ pub struct Session(SessionImp);
 // TODO: UserKnownHostsFile for custom known host fingerprint.
 
 impl Session {
+    /// The method for creating a [`Session`] and externally control the creation of TempDir.
+    ///
+    /// By using the built-in [`SessionBuilder`] in openssh, or a custom SessionBuilder,
+    /// create a TempDir.
+    ///
+    /// # Examples
+    ///
+    /// ```ignore
```

Comment:
I'm ok with `no_run` but not ok with ignoring the example altogether:

```suggestion
    /// ```no_run
```

Reply:
I will write an example to modify it